lua/better_escape.lua

```diff
@@ -49,73 +51,76 @@ local function clear_mappings()
 end
 
 -- WIP: move this into recorder.lua ?
-local last_key = nil
+-- When a first_key is pressed, `recorded_key` is set to it 
+-- (e.g. if jk is a mapping, when 'j' is pressed, `recorded_key` is set to 'j')
+local recorded_key = nil 
 local bufmodified = false
-local sequence_timer = uv.new_timer()
-local recorded_key = false
-local function log_key(key)
+local timeout_timer = uv.new_timer()
+local has_recorded = false -- See `vim.on_key` below
+local function record_key(key)
+    timeout_timer:stop()
     bufmodified = vim.bo.modified
-    last_key = key
-    recorded_key = true
-    sequence_timer:stop()
+    recorded_key = key
+    has_recorded = true 
     M.waiting = true
-    sequence_timer:start(settings.timeout, 0, function()
+    timeout_timer:start(settings.timeout, 0, function()
         M.waiting = false
-        if last_key == key then
-            last_key = nil
-        end
+        recorded_key = nil
     end)
 end
 
 vim.on_key(function()
-    if recorded_key then
-        recorded_key = false
+    if has_recorded == false then
+        -- If the user presses a key that doesn't get recorded, remove the previously recorded key.
+        recorded_key = nil
         return
     end
-    last_key = nil
+    has_recorded = false
 end)
 
--- list of modes that press <backspace> when escaping
+-- List of keys that undo the effect of pressing first_key
 local undo_key = {
     i = "<bs>",
     c = "<bs>",
     t = "<bs>",
-    v = "",
-    s = "",
 }
-local parent_keys = {}
+
+local sequences = {
+    -- Stores a sequence with this layout: mode[s] = { second_key[s] = { first_key[s] } } 
+}
```

Comment:
```suggestion
 -- Stores a sequence with this layout: mode[s] = { second_key[s] = { first_key[s] } } 
local sequences = {}
```
weird to put that comment inside of that